skops/card/_markup.py

```diff
@@ -0,0 +1,338 @@
+        """Temporarily increment indentation by one"""
+        self._indent_trace.append(spaces)
+        yield
+        self._indent_trace.pop(-1)
+
+    def _get_indent(self, *, incr: int = 0) -> str:
+        """Get current indentation, optionally incremented"""
+        # TODO: explain why skipping 1st item
+        return " " * (incr + sum(self._indent_trace[:-1]))
+
+    @staticmethod
+    def _space(value) -> str:
+        return " "
+
+    def _plain(self, value) -> str:
+        parts = [self.__call__(subitem) for subitem in value]
+        return "".join(parts)
+
+    @staticmethod
+    def _str(value) -> str:
+        # escape \
+        return value.replace("\\", "\\\\")
+
+    def _strong(self, value) -> str:
+        parts = ["**"]
+        parts += [self.__call__(subitem) for subitem in value]
+        parts.append("**")
+        return "".join(parts)
+
+    def _emph(self, value) -> str:
+        parts = ["_"]
+        parts += [self.__call__(subitem) for subitem in value]
+        parts.append("_")
+        return "".join(parts)
+
+    def _strikeout(self, value) -> str:
+        parts = ["~~"]
+        parts += [self.__call__(subitem) for subitem in value]
+        parts.append("~~")
+        return "".join(parts)
+
+    @staticmethod
+    def _raw_inline(value) -> str:
+        _, line = value
+        return line
+
+    def _raw_block(self, item) -> str:
+        # throw away the first item, which is just something like 'html'
+        # might have to revisit this if output != markdown
+        _, line = item
+        return line
+
+    def _soft_break(self, value) -> str:
+        incr = 0 if not self._indent_trace else self._indent_trace[-1]
+        return "\n" + self._get_indent(incr=incr)
+
+    def _make_content(self, content):
+        parts = []
+        for item in content:
+            part = "".join(self.__call__(item))
+            parts.append(part)
+        return "".join(parts)
+
+    def _para(self, value: list[dict[str, str]]) -> str:
+        content = self._make_content(value)
+        return content
+
+    def _header(self, value: tuple[int, Any, list[dict[str, str]]]) -> str:
+        level, _, content_parts = value
+        section_name = self._make_content(content_parts)
+        return section_name
+
+    def _image(self, value) -> str:
+        (ident, _, keyvals), caption, (dest, typef) = value
+        # it seems like ident and keyvals are not relevant for markdown
+
+        if not caption:
+            # not sure if this can be reached, just to be safe
+            raise ValueError("Figure missing a caption")
+
+        if not typef.startswith("fig:"):
+            # not sure if this can be reached, just to be safe
+            raise ValueError(f"Cannot deal with figure of type '{typef}'")
+
+        caption = "".join(self.__call__(i) for i in caption)
+        content = f"![{caption}]({dest})"
+        return content
+
+    def _figure(self, value) -> str:
```

Comment:
Yep, `Figure` is a new type that was added in pandoc 3.0. I didn't want to switch to 3.0 for CI because it is freshly released, so probably almost no one is using it. And having pandoc 2.x + 3.x in the CI matrix seemed like overkill to me.